src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -595,33 +671,125 @@ private static VersionInfo GetVersionInfo()
     internal static class BinaryPredictionTransformer
     {
         public const string LoaderSignature = "BinaryPredXfer";
+        private const string DirModel = "Model"; // This should match "DirModel" in PredictionTransformerBase<>
+
+        public static object Create(IHostEnvironment env, ModelLoadContext ctx)
+        {
+            // Load internal model to be used as TModel of BinaryPredictionTransformer<TModel>
```

Comment:
These methods appears to be duplicated, can they be refactored to avoid so much duplication?

Reply:
I can create a new class (perhaps named 'CreatePredictionTransformer') to hold an static Create Method which all the other Create methods call.

Reply:
I've refactored this into the CreatePredictionTransformer static class I talked about.

Reply:
In my latest update I've refactored this.